### ISSUE TITLE
Custom function formulas do not re-register automatically on add-in reload

### ISSUE THREAD

**Issue by hcheg** [OfficeDev/office-js#1107]
This is rather an inconvenience during development as it doesn't seem to be a problem when not hot reloading.

Edit: This become more apparent once I migrated to a shared runtime env and started using window.sharedState to share data vs the OfficeRuntimeStorage.

Once loaded, custom function formulas in Excel do not re-attach or re-register when the add-in runtime is reloaded (or hot reloaded during development).

If this behavior will not be changed, is there a way for me to force re-registration of all custom formulas in the sheet such as using something similar to:
ctx.workbook.application.calculate("FullRebuild");

## Expected Behavior

On taskpane reload, custom functions need to re-register automatically.

## Current Behavior

When a saved document containing a custom function is opened, the relevant function gets triggered, this is the expected and correct behavior.
However, if the taskpane/runtime gets reloaded, particularily during developement since users aren't able to reload once the add-in is deployed, the existing custom functions do not get triggered again.

## Steps to Reproduce, or Live Example

* Link to live example: ______

* Additional details:
1. Have a working add-in with a custom function, say, '=CONTOSO.myFunction("value1")' in cell A1.
2. Save, close then open the excel document - the associated function gets triggered (good).
3. Reload add-in without closing the document (browser devtools reload or top right menu in taskpane).
4. The associated function is not triggered (not good).
5. 
  a. If you now manually enter '=CONTOSO.myFunction("value2")' in another cell, the function again triggers successfully.
  b. Otherwise, if you re-insert the add-in (ribbon > My add-ins > myAddin), the runtime reloads automatically and now both cells trigger normally.

## Context

My app has a state management that I want the custom functions to attach to and update on state change.
Here's a little snippet:
```
function myFunction(k, invocation) { // streaming function

  console.log("triggered", k);

  // set value immediately if it exists already
  invocation.setResult(myState.getValue(k));

  // add listener
  let uuid = myState.addListener(k, v => invocation.setResult(v));

  // remove listener on cancel
  invocation.onCanceled = () => {
    myState.removeListener(k, uuid);
  };
}
```

## Your Environment

* Platform [PC desktop, Mac, iOS, Office Online]: Mac
* Host [Excel, Word, PowerPoint, etc.]: Excel
* Office version number: 16.32
* Operating System: Mac OS
* Browser (if using Office Online): NA

## Useful logs

- [ ] Console errors
- [ ] Screenshots
- [ ] Test file (if only happens on a particular file)

Thank you all for the fantastic work you've done with Office JS!
Cheers!
Hassan

**Comment by RuoyingLiang** [comment]
Thanks for reporting. The feature crew will discuss the behavior and back to this thread.

**Comment by RuoyingLiang** [comment]
@hcheg Currently the behavior is by design. But we will log your request. Could you try CTRL+SHIFT+ATL+F5 once you need to reload custom functions?

**Comment by hcheg** [comment]
Thanks @RuoyingLiang.

CTRL+SHIFT+ATL+F5 did nothing in my case, perhaps this doesn't work on Macs?

Anyway, I have found a work around, it isn't pretty but it work; and since users won't be manually reloading the add-in, the current behaviour should be fine once deployed.

For dev, I've added a 2nd arguments to CFs where the second argument simply returns a string of random characters for "cache busting": `=CONTOSO.myFunction("value1", CONTOSO.RANDCHAR())`. NOW() and RAND(), fire continuously and  so are not suitable here.
Then when the add-in reloads, `ctx.workbook.application.calculate("FullRebuild")` takes care of updating the sheet which causes the CFs to "re-register".

I understand the logic behind the current design but, coupled with good best practices guidelines, I don't believe there is any harm in allowing add-in devs to have control over CF initialization.

Thank you for your prompt response to my bug report!

Cheers,
Hassan
